molgenis/molgenis-emx2
# feat(graphql): can expand into the fields of subclasses

N.B. best wait until #3293 has been completed

how to test:
- assume you have a data model two tables: Person and Employee extends Person.
- in old situation of you query Person you will get also Employee but not employee specific columns
- in new situation you can also query the employee specific columns

N.B. this is only in the query, not the filters or mutations.

## How to test

1. Create a new schema and upload this test schema: [subclass_example.xlsx](https://github.com/user-attachments/files/20885349/subclass_example.xlsx). It has the following tables: teams, persons, employees, and managers. Employees and Managers extend Persons.
2. Run this query. Data in the inherited table should be visible. For each team, I would like to employees and managers, as well as the data that's specific to each table.

```graphql
{
  Teams {
    team
    members {
      mg_tableclass
      firstName
      lastName
      age
      gender
      employeeId
      role
      managerOfTeam {
        team
      }
      managerSince
    }
  }
}
```


todo:
- [x] updated docs in case of new feature (will be in graphql generated docs, do we need more?)
- [x] we need way more test cases?
- [ ] check if there are issues with catalogue data model that affect this PR
